service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -183,13 +188,13 @@ public static DigitalTwinAsyncClient createFromConnectionString(String connectio
             options = new DigitalTwinInvokeCommandRequestOptions();
         }
 
-        Object payloadAsObject = null;
-        if(payload != null)
+        // Retrofit does not work well with null in body
+        if(payload == null)
         {
-            payloadAsObject = objectMapper.readValue(payload, Object.class);
+            payload = "";
         }
 
-        return digitalTwin.invokeRootLevelCommandWithServiceResponseAsync(digitalTwinId, commandName, payloadAsObject, options.getConnectTimeoutInSeconds(), options.getResponseTimeoutInSeconds())
+        return digitalTwin.invokeRootLevelCommandWithServiceResponseAsync(digitalTwinId, commandName, payload, options.getConnectTimeoutInSeconds(), options.getResponseTimeoutInSeconds())
```

Comment:
You're now sending the payload as a string as opposed to as an Object. Is this intentional?

Reply:
We have some issues with serialization. I was working on a fix for it and it is ready now. I will send it out in the next PR so that you can understand the difference.

Reply:
https://github.com/Azure/azure-iot-sdk-java/pull/948